syntax = "proto3";

// The greeting service definition.
service greeter {
  // Sends a greeting
  rpc say_hello (HelloRequest) returns (HelloReply) {}
  rpc say_hello_goodbye (HelloRequest) returns (stream HelloReply) {}
  rpc say_hello_to_many (stream HelloRequest) returns (stream HelloReply) {}
  rpc say_hello_to_many_at_once (stream HelloRequest) returns (HelloReply) {}
}

// The request message containing the user's name.
message HelloRequest {
  string name = 1;
}



// The response message containing the greetings
message HelloReply {
  string message = 1;
}

